content/docs/concepts/features/healing.md

```diff
@@ -0,0 +1,50 @@
+<script type="text/javascript" src="https://viewer.diagrams.net/js/viewer-static.min.js"></script>
+
+## Description
+
+ Healing is a crucial functionality of the NSM, designed to ensure the high availability and resilience of workloads using NSM. This feature leverages automation and self-healing mechanisms to detect and address issues that may impact the reliability and performance of network services deployed on the environment.
+
+## Benefits
+
+**Enhanced Reliability:** healing enhances the reliability of applications by automatically recovering from failures, reducing manual intervention, and minimizing downtime. \
+**Improved User Experience:** The feature ensures a consistent user experience by maintaining service availability, even when underlying infrastructure or application components experience issues. \
+**Efficient Resource Utilization:** Through proactive scaling and resource management, healing optimizes resource utilization, ensuring efficient NSM domain operation. \
+**Simplified Operations:** Healing simplifies operational tasks, reducing the burden on administrators and DevOps teams. \
+
+## Conception
+
+The main conception of the healing in NSM is pretty simple and can be described with next facts:
+
+1. Healing is managing only and only on the Network Service Client.
+2. Network Service Client uses `monitor connections` api to monitor its NSMgr to keep the connection up to date.
+3. If a Network Service Client gets an event from the NSMgr with changing state, or deleting connection, or stream closing then client forces a new Request.
+4. A Network Service Client should close existing an connection and request a new one only and only if connectivity is gone.
+
+And that's it!
```

Comment:
You may want to talk about how events originate and propogate back through the chain.

Reply:
Currently, NSM monitoring exists separately from healing, and it could be used flexiblely. The healing is only one example of how an NSM monitoring API could be used.

So I think we could create a separate concept about monitroing and add links from monitoring to healing and vice versa.

This approach could reduce the complexity of the healing concept and also clarify monitoring in detail.

Thoughts?